setup.py

```diff
@@ -66,11 +66,11 @@ def read(filename):
     ],
     extras_require={
         "notebook": [
-            # Notebook upper limit is a workaround for issues with IPython not being defined.
-            "notebook>=4.4.1,<7",
+            "notebook>=7",
             "ipywidgets>=7.0.0",
-            "jupyter_client<8",  # https://github.com/jupyter/notebook/issues/6748
-            "pyzmq<25",  # https://github.com/jupyter/notebook/issues/6748
+            "jupyter_client>=8",
+            "ipympl",
+            "pyzmq",
```

Comment:
Is `pyzmq` even needed here anymore? It's not a direct dependency and we don't pin it. So probably doesn't even need to be in the list.

Reply:
Nope. Removing...